src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1246,10 +1008,9 @@ loadByHash(uint256 const& ledgerHash, Application& app, bool acquire)
 {
     std::shared_ptr<Ledger> ledger;
     {
-        std::ostringstream s;
-        s << "WHERE LedgerHash = '" << ledgerHash << "'";
-        std::tie(ledger, std::ignore, std::ignore) =
-            loadLedgerHelper(s.str(), app, acquire);
+        std::optional<LedgerInfo> info =
+                app.getSQLInterface().loadLedgerInfoByHash(ledgerHash);
+        if (info) ledger = loadLedgerHelper(*info, app, acquire);
```

Comment:
What do you think of refactoring this as follows:
```
std::shared_ptr<Ledger>
loadByHash(uint256 const& ledgerHash, Application& app, bool acquire)
{
    if (std::optional<LedgerInfo> info =
            app.getSQLInterface().loadLedgerInfoByHash(ledgerHash))
    {
        std::shared_ptr<Ledger> ledger = loadLedgerHelper(*info, app, acquire);
        finishLoadByIndexOrHash(ledger, app.config(), app.journal("Ledger"));
        assert(!ledger || ledger->info().hash == ledgerHash);
        return ledger;
    }
    return {};
}
```

Reply:
Done.